src/groupByKey.ts

```diff
@@ -2,6 +2,19 @@
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey(
+  items: Array<string | number>,
+  key: string | number,
+): GroupsMap {
+  const result: GroupsMap = {};
+
+  for (const item of items) {
+    if (result.hasOwnProperty(item[key])) {
+      result[item[key]].push(item);
+    } else {
+      result[item[key]] = [item];
```

Comment:
The expression `item[key]` is used multiple times. Consider storing it in a variable before the `if` statement to avoid repeating the same property access and to improve code readability.